File: unique.2.js
function unique(string){
    const dictionary = {};
    const array = string.split("");

    for(let i = 0; i < array.length; i++){
        if(dictionary[array[i]])
            return false;

        dictionary[array[i]] = true;
    }

    return true;
}

console.log(unique("abcdefghijklmnopqrstuvwxyz"));
console.log(unique("abcdefghijklmnopqrstuvwxyzz"));
console.log(unique("aa"));
console.log(unique("a"));
console.log(unique(""));

File: unique.2.go
package main

import "fmt"

func unique(input string) bool {
	dictionary := map[rune]bool{}

	for _, char := range input {
		if _, ok := dictionary[char]; ok {
			return false
		}

		dictionary[char] = true
	}

	return true
}

func main() {
	fmt.Println(unique("abcdefghijklmnopqrstuvwxyz"))
	fmt.Println(unique("abcdefghijklmnopqrstuvwxyzz"))
	fmt.Println(unique("aa"))
	fmt.Println(unique("a"))
	fmt.Println(unique(""))
}
